### PR TITLE
Add `.rspec` with `--warnings`.

### PR DESCRIPTION
**Pull request by ioquatix**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ioquatix** [comment]
https://github.com/gettalong/kramdown/issues/634

**Comment by ioquatix** [comment]
```
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
/Users/samuel/Documents/ioquatix/aws-sdk-ruby/gems/aws-sdk-core/lib/aws-sdk-core/instance_profile_credentials.rb:187: warning: Net::HTTPResponse#header is obsolete
```

Might require some more attention - beyond my time availability.

**Comment by ioquatix** [comment]
Unfortunately https://github.com/aws/aws-sdk-ruby/pull/2179 has been held up for almost 2 years. But it's responsible for ~20k warnings when running specs.

**Comment by ioquatix** [comment]
Please ensure you merge the changes to `.rspec` this time. https://github.com/aws/aws-sdk-ruby/issues/1778#issuecomment-434106447 said there would be a follow up but there never was. Because as you can see, without constant attention, it's easy for new warnings to show up. There are many issues where people are submitting fixes for warnings, but they should never have occurred in the first place. Thanks.

File: gems/aws-sdk-autoscaling/lib/aws-sdk-autoscaling/activity.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ def initialize(*args)
       @id = extract_id(args, options)
       @data = options.delete(:data)
       @client = options.delete(:client) || Client.new(options)
+      @waiter_block_warned = false
```
Comment (mullermp) [inline gems/aws-sdk-autoscaling/lib/aws-sdk-autoscaling/activity.rb]: These service files are generated. I see that you added this to the mustache template as well. Were these added manually or the result of `rake build`?

**Comment by mullermp** [comment]
@ioquatix Sorry that it was dropped. I can merge this, but I'd like to do so after re:Invent to avoid any hidden problems. I hope that's okay. I have changes queued up for next week and I'd like to minimize any impact.

File: gems/aws-sdk-autoscaling/lib/aws-sdk-autoscaling/activity.rb (hunk 1 above)
Comment (ioquatix) [reply gems/aws-sdk-autoscaling/lib/aws-sdk-autoscaling/activity.rb]: `rake build`

**Comment by mullermp** [comment]
Everything caught on fire today anyway, so I'll just go and merge this now. Thanks again for the contribution.

**Comment by mullermp** [comment]
@ioquatix I am reverting the --warnings change. While squashing warnings is helpful (I will run it from time to time), we are unfortunately pinned to an old version of mustache and it makes it difficult to diagnose spec failures and slows down CI. We're trying to get off mustache and instead use Smithy (https://awslabs.github.io/smithy/) but that work would be a version 4 that only follows a version 2 deprecation.

**Comment by ioquatix** [comment]
Mustache is releasing an update hopefully fixing all warnings.

**Comment by mullermp** [comment]
@ioquatix is the update to the older version that supports Ruby 1.9.3?

Gemfile:
```
  gem 'mustache', '0.99.8' # pinned to support Ruby 1.9.3
```

Someone would need to create a branch (such as v0.99.9) starting at v0.99.8, fix the warnings, and then publish. We could change the pinned dependency to ~> 0 because version 1.0.0 is ruby > 2. Going forward, if there were more things to fix, it would need to have a version such as 0.99.9.1, 0.99.9.2, etc.

**Comment by mullermp** [comment]
Adding to this, I would LOVE to drop Ruby 1.9.3 support. As well as deprecating V2. I'll start those discussions internally after re:Invent.

**Comment by ioquatix** [comment]
@mullermp Do you have an update?

**Comment by mullermp** [comment]
I was able to get leadership on board with deprecating version 2 (blog: https://aws.amazon.com/blogs/developer/deprecation-schedule-for-aws-sdk-for-ruby-v2/)

Runtimes are still an open question - I think that there will be some global SDK maintenance policy in the future that will allow us to make those decisions. Based on the data, the usage of V3 on Ruby 1.9->2.2 is very low and I want to make that case for dropping support.

**Comment by ioquatix** [comment]
Awesome! Thanks for the update.

**Pull request merged**